apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -699,51 +699,47 @@ func (m *ModelManagerAPI) ListModelsWithInfo(user params.Entity) (params.ModelIn
 	}
 
 	for _, mi := range modelInfos {
-		info := params.ModelInfoResult{
-			Result: &params.ModelInfo{
-				Name:           mi.Name,
-				UUID:           mi.UUID,
-				OwnerTag:       names.NewUserTag(mi.Owner).String(),
-				ControllerUUID: mi.ControllerUUID,
-				Life:           params.Life(mi.Life.String()),
-
-				CloudTag:           mi.CloudTag,
-				CloudRegion:        mi.CloudRegion,
-				CloudCredentialTag: mi.CloudCredentialTag,
-
-				SLA: &params.ModelSLAInfo{
-					Level: mi.SLALevel,
-					Owner: mi.Owner,
-				},
-
-				DefaultSeries: mi.DefaultSeries,
-				ProviderType:  mi.ProviderType,
-				AgentVersion:  mi.AgentVersion,
-
-				Status: common.EntityStatusFromState(mi.Status),
+		summary := &params.ModelSummary{
+			Name:           mi.Name,
+			UUID:           mi.UUID,
+			OwnerTag:       names.NewUserTag(mi.Owner).String(),
+			ControllerUUID: mi.ControllerUUID,
+			Life:           params.Life(mi.Life.String()),
+
+			CloudTag:           mi.CloudTag,
+			CloudRegion:        mi.CloudRegion,
+			CloudCredentialTag: mi.CloudCredentialTag,
+
+			SLA: &params.ModelSLAInfo{
+				Level: mi.SLALevel,
+				Owner: mi.Owner,
+			},
+
+			DefaultSeries: mi.DefaultSeries,
+			ProviderType:  mi.ProviderType,
+			AgentVersion:  mi.AgentVersion,
+
+			Status: common.EntityStatusFromState(mi.Status),
+			Counts: []params.ModelEntityCount{},
+			User: params.ModelUserInfo{
+				LastConnection: mi.UserLastConnection,
 			},
 		}
-		result.Results = append(result.Results, info)
-
-		//TODO (anastasaiamac 2017-11-22) cater for
-		//mi.MachineCount
-		//mi.CoreCount
-		//mi.Access
-		//mi.UserLastConnection
-
-		// TODO (anastasaiamac 2017-11-22) should just ignore mi.Machines?
-		//  Dealing with machines
-		if len(mi.Machines) > 0 {
-			returnedMachines := []params.ModelMachineInfo{}
-			for machineId, machine := range mi.Machines {
-				// TODO (anastasiamac 2017-11-22) fill this out with details
-				returnedMachines = append(returnedMachines, params.ModelMachineInfo{
-					Id:         machineId,
-					InstanceId: machine.InstanceId,
-				})
-			}
-			info.Result.Machines = returnedMachines
+
+		if mi.MachineCount > 0 {
+			summary.Counts = append(summary.Counts, params.ModelEntityCount{params.Machines, uint64(mi.MachineCount)})
```

Comment:
I actually prefer int64. While the number shouldn't ever be negative, bad math means we end up with a more obvious "-1" rather than 2^64.